src/PhpBrew/VariantBuilder.php

```diff
@@ -158,7 +158,25 @@ public function __construct()
         $this->variants['json'] = '--enable-json';
         $this->variants['hash'] = '--enable-hash';
         $this->variants['exif'] = '--enable-exif';
-        $this->variants['mbstring'] = '--enable-mbstring';
+
+        $this->variants['mbstring'] = function (ConfigureParameters $params, Build $build, $value) {
+            $params = $params->withOption('--enable-mbstring');
+
+            if ($build->compareVersion('5.4') >= 0 && !$build->isDisabledVariant('mbregex')) {
```

Comment:
Could you find why exactly it failed, maybe a few lines above in the log? It works fine for me:
```
./configure '--cache-file=/home/morozov/.phpbrew/cache/config.cache' '--prefix=/home/morozov/.phpbrew/php/php-5.6.40' '--with-config-file-path=/home/morozov/.phpbrew/php/php-5.6.40/etc' '--with-config-file-scan-dir=/home/morozov/.phpbrew/php/php-5.6.40/var/db' '--with-libdir=lib/x86_64-linux-gnu' '--with-bz2=/usr' '--with-curl=/usr' '--enable-fpm' '--with-gd=shared' '--with-jpeg-dir=/usr' '--with-png-dir=/usr' '--with-freetype-dir=/usr' '--with-iconv' '--enable-mbstring' '--with-onig=/usr' '--with-mysql=mysqlnd' '--with-mysqli=mysqlnd' '--with-mysql-sock=/var/run/mysqld/mysqld.sock' '--with-openssl=/usr' '--with-sqlite3' '--enable-soap' '--with-zlib=/usr' '--enable-dom' '--enable-libxml' '--with-libxml-dir=/usr' '--enable-simplexml' '--enable-xml' '--enable-xmlreader' '--enable-xmlwriter' '--with-xsl' '--enable-opcache' >> '/home/morozov/.phpbrew/build/php-5.6.40/build.log' 2>&1
===> Checking patches...
Checking patch for php5.3.29 multi-sapi
Checking patch for php5.3.x on 64bit machine when intl is enabled.
Checking patch for openssl dso linking patch
Checking patch for php5.6 with openssl 1.1.x patch.
---> Applying patch...
patching file ext/openssl/openssl.c
patching file ext/openssl/xp_ssl.c
patching file ext/phar/util.c
patching file ext/openssl/openssl.c
patch unexpectedly ends in middle of line
Hunk #1 succeeded at 651 with fuzz 1.

Done.
1 changes patched.
===> Building...
make >> '/home/morozov/.phpbrew/build/php-5.6.40/build.log' 2>&1
Build finished: 4.7 minutes.
```

Reply:
Here comes my phpbrew command and log

**command**
```
phpbrew --debug install --jobs "$(nproc)" 5.6.40 +default +dbs +dba +apxs2 +ftp +soap +gettext +exif +iconv +bz2 +zli
```

**log**
https://gist.github.com/markwu/520647ce338f3d8a0543a731e20c8152

Reply:
I only tried to build PHP 7.4 against Homebrew dependencies, and it worked for me as well. Judging by the log output, there might be an older version of oniguruma required.

Reply:
If it turns out that this is because of the incompatible version, we can either find a way to enforce the version of the library used to build PHP 5.6 or reopen #1132 and use the external library only if it's required (PHP 7.4).

Reply:
Okay, I think I found the problem.

PHP-5.6.40 and PHP-7.3.13 ships it's own `oniguruma`, but PHP-7.4.1 doesn't. It located under

`~/.phpbrew/build/php-5.6.40/ext/mbstring/oniguruma`
`~/.phpbrew/build/php-7.3.13/ext/mbstring/oniguruma`

I guess PHP-7.(0|1|2) ships with it's own`oniguruma`, too.

But, PHP-5.6.40 can only build `--enable-mbregex` with it's shipped `oniguruma`, and php-7.3.13 can build `--enable-mbregex` with both shipped and external `oniguruma`

So, as you said, maybe we should reopen https://github.com/phpbrew/phpbrew/pull/1132 to force build previous PHP version with it's own shipped `oniguruma`.

Reply:
That's why homebrew's php formula has additional `oniguruma` dependency after PHP-7.4.0.